### PR TITLE
Add gas payment to transferRemote and upgrade to SDK 1.2.0

### PR DESCRIPTION
**Pull request by jmrossy**
Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1680

- Update to SDK 1.2.0
- Use gas payment in warp route
- Fix minor UI bugs

### CODE REVIEW

File: src/features/transfer/useTokenTransfer.ts [hunk 1]
```diff
@@ -83,16 +84,20 @@ export function useTokenTransfer(onDone?: () => void) {
...
           ? getHypErc20CollateralContract(tokenRoute.hypCollateralAddress, provider)
           : getHypErc20Contract(tokenRoute.sourceTokenAddress, provider);
+        const gasPayment = await hypErc20.quoteGasPayment(destinationChainId);
         const transferTxRequest = await hypErc20.populateTransaction.transferRemote(
           destinationChainId,
           utils.addressToBytes32(recipientAddress),
           weiAmount,
+          {
+            value: gasPayment,
+          },
```
Comment (yorhodes) [inline src/features/transfer/useTokenTransfer.ts:101]: should the SDK generally be returning results from `populateTransaction`?

Comment (jmrossy) [reply src/features/transfer/useTokenTransfer.ts:101]: Would be good yeah. Having both options (`prepareTransaction` and `sendTransaction`) would be ideal

File: src/features/transfer/useTokenTransfer.ts [hunk 2]
```diff
@@ -83,16 +84,20 @@ export function useTokenTransfer(onDone?: () => void) {
         const hypErc20 = isNativeToRemote
           ? getHypErc20CollateralContract(tokenRoute.hypCollateralAddress, provider)
           : getHypErc20Contract(tokenRoute.sourceTokenAddress, provider);
+        const gasPayment = await hypErc20.quoteGasPayment(destinationChainId);
```
Comment (nambrot) [inline src/features/transfer/useTokenTransfer.ts:94]: Flagging here that this method does not exist on the audited method, so we will probably want to make another PR for that version and then tag that as the audit-compliant version of the UI
